libopenage/datastructure/doubly_linked_list.h

```diff
@@ -114,28 +118,42 @@ class DoublyLinkedList {
 
 	/**
 	 * Fully delete a node and fix the neighbor pointers.
+	 * Update #first or #last, if necessary.
 	 * O(1)
+	 * @pre \p node belongs to this list.
 	 */
 	void erase(node_t *node) {
```

Comment:
Do we want the erase() functions to correct the first/last pointer if necessary? 
there can be a protected unlink() function, which does the same without setting first and last.